### ISSUE TITLE
Recipes appear only after rejoin

### ISSUE THREAD

**Issue by 123oro321** [Buuz135/Thaumic-JEI#61]
I have noticed when you have the hide recipes if missing research option enabled, after you make a new research JEI won't show the new recipes, but only after the player rejoins the world.
It does it the other way too - if i use command to revoke a research, the recipes will still appear till i rejoin.
Thaumcraft version: 6.1.BETA26
JEI version: 4.15.0.291
ThaumicJEI version: 1.6.0-27

**Comment by xJon** [comment]
I believe this is a duplicate of https://github.com/Buuz135/Thaumic-JEI/issues/34.
